woocommerce/woocommerce-ios
# [Order Details] Do not show Mark Order Complete for unpaid orders

Closes: #7207 

### Description

Unpaid orders and orders in the Processing status currently show the `Mark Order Complete` button prominently as the first and main call to action. They also show `Collect Payment` when no date_paid is set for the order, which may cause confusion.

If the merchant taps "Mark Order Complete" the side effect is that the order gets marked as paid, but a better primary CTA is to use the `Collect Payment` button to show the payment options. If they need to, the merchant can still perform the mark complete action by using the pencil icon next to the status.

We want to accomplish the following:

1. For unpaid orders (`.pending`, `.onhold`. `.failed`): Do not show the "Mark Order Complete" button. 
2. For paid orders (`.complete`, `.processing`): Show the "Mark Order Complete" button as primary CTA.
3. For paid orders (`.complete`, `.processing`), if Shipping Labels are available. : Show the `Create Shipping Label` as primary CTA, as well as the "Mark Order Complete`button as a secondary CTA.

This PR addresses these items by:

- Rename `isProcessingPayment`to `isProcessingStatus` to avoid confusion. `OrderStatusEnum.processing` means that the order is paid.
- Adds `isPendingPayment` check for `OrderStatusEnum.pending`. Tne order is still unpaid.
- Changes `isEligibleForPayment` logic: Previously we if an order was eligible for payment based on `Order.datePaid`, however, this can exist for unpaid orders if we switch the Order Status. Now is based directly on Order Status.

**Scenarios:**
| Order | Status | Shipping Labels? | Buttons |
|---|---|---|---|
| Paid | `.completed` | No | -
| Paid | `.completed` | Yes | `Create Shipping Label`|
| Paid | `.processing` | No | `Mark as Completed` |
| Paid | `.processing` | Yes | `Mark as Completed` `Create Shipping Label` |
| Unpaid | `.pending` `.failed` | - | `Collect Payment`|
| Draft/Cancelled/Refunded/Hold | `.autodraft`, `.cancelled`, `.refunded`, `on hold`| - | - |

### Testing instructions
- Enable [WooCommerce Shipment Tracking](https://woocommerce.com/products/shipment-tracking/)
- By checking existing Orders, or creating a new Order under Orders > "+" we can check the different scenarios above:

1. Case 1: Paid order. Marked as Complete. No Shipping Label (Digital Product) -> No buttons.
2. Case 2: Paid order. Marked as Complete. Shipping Label (Shippable Product) -> `Create Shipping Label` button.
3. Case 3: Paid order. Marked as Processing. No Shipping Label (Digital Product) -> `Mark as Completed` button as primary CTA
4. Case 4: Paid order. Marked as Processing. Shipping Label (Shippable Product) -> `Mark as Completed` as secondary CTA and `Create Shipping Label` button as primary CTA.
5. Case 5: Unpaid order. Marked as Pending, or Failed -> `Collect Payment` button
6. Case 6: Draft/Cancelled/Refunded order. Marked as Pending, On-Hold, or Failed -> No buttons.

| Case 1:  | Case 2: | Case 3: | 
|---|---|---|
|![_case 1 complete   non-shippable](https://user-images.githubusercontent.com/3812076/178638237-ecb3cf5e-7f76-48e9-8e9e-8fe157604513.png)| ![_case 2 completed   shipping](https://user-images.githubusercontent.com/3812076/178638261-65817cca-df74-4ba5-891b-b3b456e18031.png) |![_case 3 - digital and processing, mark complete 1st CTA](https://user-images.githubusercontent.com/3812076/178638279-decc1ed0-187f-4d94-bd66-c45f3c2b877a.png)  |

| Case 4: | Case 5: | Case 6: |
|---|---|---|
| ![_case 4 paid, mark complete secondary cta](https://user-images.githubusercontent.com/3812076/179192483-35e6cb5e-d184-4cf0-ae86-077c04f8fe74.png) | ![_case 5   failed](https://user-images.githubusercontent.com/3812076/179193774-76b59a49-113f-4b22-80bd-33efcabd152d.png) | ![_case 6 refunded show no buttons](https://user-images.githubusercontent.com/3812076/178638318-f6fc2235-3767-450c-bd02-76bcace91a23.png) |


### Screenshots


---
